### PR TITLE
Made some new things

### PR DESCRIPTION
**Pull request by Scraayp**
Added a option to enable and disable modlog

With a option to set a modlog channel with the name
Added a option to enable and disabled remove messages.
-- It will only trigger on the treshold of a punishment.
-it will clear the chat will the treshold of the punishment that got triggerd

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,62 +1,63 @@
...
+	maxDuplicatesWarning: 7, // Amount of duplicate messages that trigger a warning.
+	maxDuplicatesKick: 10, // Amount of duplicate messages that trigger a warning.
+	maxDuplicatesBan: 12, // Amount of duplicate messages that trigger a warning.
+	maxDuplicatesMute: 9, // Amount of duplicate messages that trigger a warning.
+	// Discord permission flags: https://discord.js.org/#/docs/main/master/class/Permissions?scrollTo=s-FLAGS
+	exemptPermissions: [ 'ADMINISTRATOR'], // Bypass users with any of these permissions(These are not roles so use the flags from link above).
+	ignoreBots: true, // Ignore bot messages.
+	verbose: true, // Extended Logs from module.
+	ignoredUsers: [], // Array of User IDs that get ignored.
+	removeMessages: true, // Array of User IDs that get ignored.
```
Comment (Androz2091) [inline README.md:46]: Description should be changed.

File: index.js [hunk 2]
```diff
@@ -1,525 +1,559 @@
...
+ * @property {Array<string>|function} [ignoredRoles=[]] Array of string role IDs or role name that are ignored
+ * @property {Array<string>|function} [ignoredGuilds=[]] Array of string Guild IDs that are ignored
+ * @property {Array<string>|function} [ignoredChannels=[]] Array of string channels IDs that are ignored
+ *
+ * @property {boolean} [warnEnabled=true] If false, the bot won't warn users
+ * @property {boolean} [kickEnabled=true] If false, the bot won't kick users
+ * @property {boolean} [banEnabled=true] If false, the bot won't ban users
+ * @property {boolean} [muteEnabled=true] If false, the bot won't mute users
+ * 
+ * @property {boolean} [modlog=false] If false, the bot won't send moderation logs in a specefic channel.
```
Comment (Androz2091) [inline index.js:57]: Here it's `modlog` and line 95 it's `modlogEnabled`. I think the best solution is to set `modLogChannel` to `null` by default. And if it has a value, we can consider moderation logs enabled.

File: index.js [hunk 3]
```diff
@@ -1,525 +1,559 @@
...
+				await message.member.ban({
+					reason: 'Spamming!',
+					days: options.deleteMessagesAfterBanForPastDays
+				});
+				if (options.banMessage)
+					await message.channel.send(format(options.banMessage, message)).catch(e => {
+						if (options.verbose) console.error(e);
+					});
+					if(options.modlogEnabled)
+					let modlogCh = message.guild.channels.cache.find(ch => ch.name == options.modLogChannel)
```
Comment (Androz2091) [inline index.js:239]: Here we should support v11 of Discordjs. Also, we should be able to find the channel using ID. (these changes should be applied everywhere)
```suggestion
					const modlogCh = (v11 ? message.guild.channels : message.guild.channels.cache).find(ch => ch.name === options.modLogChannel || ch.id === options.modLogChannel)
```

File: index.js [hunk 4]
```diff
@@ -1,525 +1,559 @@
...
+					if(modlogCh){
+						modlogCh.send(`Spam detected: ${message.author} got **banned**`)
```
Comment (Androz2091) [inline index.js:241]: Add missing semicolon.
```suggestion
						modlogCh.send(`Spam detected: ${message.author} got **banned**`);

```

**Comment by Scraayp** [comment]
@Androz2091 Everything you requested got made.
You can review it again

**Comment by MrAugu** [comment]
@Androz2091 you could release a semi-major release and ditch the support for discordjs v11 anymore, since the stable is now v12.

**Comment by Androz2091** [comment]
Even if the stable is now v12, many, many people are still using v11. We need, at least, to have a support-v11 branch or something like that.

**Comment by Scraayp** [comment]
Can we leave that to discord. And now just review the pr

:)

**Comment by Androz2091** [comment]
🙄 Please be patient.

**Comment by Scraayp** [comment]
I am lol but this more off topic of the pr

**Comment by Scraayp** [comment]
I am not i meant

**Comment by MrAugu** [comment]
v11 reached it's end of life, you could warn users to install an older version that still supports v11, and future versions of discord-anti-spam to only support v12

**Comment by Scraayp** [comment]
This has been already added in the dev branch!

**Pull request closed without merging**